azurerm/resource_arm_virtual_machine_scale_set_test.go

```diff
@@ -66,16 +70,57 @@ func TestAccAzureRMVirtualMachineScaleSet_basicAcceleratedNetworking(t *testing.
 			{
 				Config: config,
 				Check: resource.ComposeTestCheckFunc(
-					testCheckAzureRMVirtualMachineScaleSetExists(resource_name),
-					testCheckAzureRMVirtualMachineScaleSetAcceleratedNetworking(resource_name, true),
+					testCheckAzureRMVirtualMachineScaleSetExists(resourceName),
+					testCheckAzureRMVirtualMachineScaleSetAcceleratedNetworking(resourceName, true),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMVirtualMachineScaleSet_basicIPForwarding(t *testing.T) {
+	resourceName := "azurerm_virtual_machine_scale_set.test"
+	ri := acctest.RandInt()
+	config := testAccAzureRMVirtualMachineScaleSet_basicIPForwarding(ri, testLocation())
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMVirtualMachineScaleSetDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMVirtualMachineScaleSetExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "ip_forwarding", "true"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMVirtualMachineScaleSet_basicDNSSettings(t *testing.T) {
+	resourceName := "azurerm_virtual_machine_scale_set.test"
+	ri := acctest.RandInt()
+	config := testAccAzureRMVirtualMachineScaleSet_basicDNSSettings(ri, testLocation())
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMVirtualMachineScaleSetDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMVirtualMachineScaleSetExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "dns_settings.0.dns_servers.0", "8.8.8.8"),
```

Comment:
You have dns_settings as a `TypeSet` so the `.0.` syntax will not work. You'll need to change to a `TypeList`

Reply:
Hello @katbyte while this would help with the test, TypeSet is the right fit here since we can only have a single `dns_settings` block. Same logic as for the `network_profile` block.

While debugging I realized that using the 

since the full string would be something like this, as you can see the `dns_settings` and the `ip_forwarding` are a child element of the `network_profile` with TypeSet, therefore I won't be able to use the `resource.TestCheckResourceAttr` logic unless you know of another method that supports sets

```
"network_profile.4108483151.ip_forwarding": "true",
```

Reply:
Typically unless there is a reason to, single element blocks like this are always a `TypeList` as diffs are simpler, easier to get at the attributes and state migrations are not required when adding/changing the fields.

Reply:
ok, while changing the `dns_settings` to `TypeList` will help, what should we do with `network_profile` ?

1) convert to `TypeList`
2) find a way to generate the hash in the test.